### PR TITLE
Format number fields (thousand separator)

### PR DESCRIPTION
**Pull request by dimak1**
Add thousand separator to number fields in the form.
Only for display purposes - actual number (without commas) will be saved to the table.

### CODE REVIEW

**Comment by dimak1** [comment]
@matthieu-foucault rebased and fixed conflicts, should be good now...

File: app/components/Application/ApplicationDetailsCardItemCustomFields.tsx [hunk 1]
```diff
@@ -75,6 +76,13 @@ const customFields = (showDiff, diffPathArray, diffArray, handleEnums) => {
     emissionGas: props => <SummaryEmissionGasFields {...props} />,
     production: props => (
       <ProductionFields query={props.formContext.query} {...props} />
+    ),
+    NumberField: props => (
+      <NumberFormat
+        thousandSeparator
+        displayType="text"
+        value={props.formData}
+      />
```
Comment (matthieu-foucault) [inline app/components/Application/ApplicationDetailsCardItemCustomFields.tsx]: This needs to handle the display of differences, the same way `StringField` does

File: app/containers/Forms/EmissionGasFields.tsx [hunk 2]
```diff
@@ -1,11 +1,30 @@
 import React from 'react';
 import {FieldProps} from 'react-jsonschema-form';
 import {Form, Col} from 'react-bootstrap';
+import {JSONSchema6} from 'json-schema';
+import NumberFormat from 'react-number-format';
 
-const EmissionGasFields: React.FunctionComponent<FieldProps> = ({
+interface Props extends FieldProps {
+  annualEmission?: any;
+}
```
Comment (matthieu-foucault) [inline app/containers/Forms/EmissionGasFields.tsx]: Doesn't look like you need this. You can just use `FieldProps` below

File: app/containers/Forms/EmissionGasFields.tsx [hunk 3]
```diff
@@ -20,31 +39,41 @@ const EmissionGasFields: React.FunctionComponent<FieldProps> = ({
...
-            type="number"
-            value={formData.annualEmission}
-            onChange={e => {
+          <registry.fields.NumberField
+            required
+            schema={annualEmissionSchema}
+            uiSchema={uiSchema.annualEmission}
+            formData={formData.annualEmission}
+            autofocus={autofocus}
+            idSchema={idSchema}
```
Comment (matthieu-foucault) [inline app/containers/Forms/EmissionGasFields.tsx]:
```suggestion
            idSchema={idSchema.annualEmission as IdSchema}
```
Or something like this, look at the updated `ProductionFields`

File: app/containers/Forms/FuelFields.tsx [hunk 4]
```diff
@@ -50,17 +63,25 @@ const FuelFields: React.FunctionComponent<Props> = ({
...
-            type="number"
-            defaultValue={formData.quantity}
-            onChange={e => {
+          <registry.fields.NumberField
+            required
+            schema={quantitySchema}
+            uiSchema={uiSchema.quantity}
+            formData={formData.quantity}
+            autofocus={autofocus}
+            idSchema={idSchema}
```
Comment (matthieu-foucault) [inline app/containers/Forms/FuelFields.tsx]: Same as above re. idSchema

File: app/containers/Forms/NumberFields.tsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import NumberFormat from 'react-number-format';
+import {FieldProps} from 'react-jsonschema-form';
+
+const NumberFields: React.FunctionComponent<FieldProps> = ({
```
Comment (matthieu-foucault) [inline app/containers/Forms/NumberFields.tsx]: Rename this (and the file to `NumberField`)

File: app/containers/Forms/NumberFields.tsx [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  formData,
+  onChange
+}) => {
+  return (
+    <NumberFormat
+      thousandSeparator
+      className="form-control"
+      allowNegative={false}
```
Comment (matthieu-foucault) [inline app/containers/Forms/NumberFields.tsx]: This should depend on the `schema` prop. More precisely the `minimum` or `exclusiveMinimum` values, if defined:
```js
const allowNegative = (schema.minimum === undefined && schema.exclusiveMinimum === undefined) || minimum < 0 || exclusiveMinimum < 0
``` 
See https://json-schema.org/understanding-json-schema/reference/numeric.html

File: app/containers/Forms/EmissionGasFields.tsx (hunk 2 above)
Comment (dimak1) [reply app/containers/Forms/EmissionGasFields.tsx]: Updated

File: app/containers/Forms/EmissionGasFields.tsx (hunk 3 above)
Comment (dimak1) [reply app/containers/Forms/EmissionGasFields.tsx]: Updated

File: app/containers/Forms/FuelFields.tsx (hunk 4 above)
Comment (dimak1) [reply app/containers/Forms/FuelFields.tsx]: Updated

File: app/containers/Forms/NumberFields.tsx (hunk 5 above)
Comment (dimak1) [reply app/containers/Forms/NumberFields.tsx]: Renamed

File: app/components/Application/ApplicationDetailsCardItemCustomFields.tsx (hunk 1 above)
Comment (dimak1) [reply app/components/Application/ApplicationDetailsCardItemCustomFields.tsx]: Updated.

File: app/containers/Forms/NumberFields.tsx (hunk 6 above)
Comment (dimak1) [reply app/containers/Forms/NumberFields.tsx]: Updated.

**Comment by dimak1** [comment]
@matthieu-foucault rebased

**Pull request merged**